architxt/simplification/tree_rewriting/__init__.py

```diff
@@ -267,14 +270,30 @@ def apply_operations(
                 early_exit,
                 simplification_operation,
                 barrier,
+                mlflow_run_id=mlflow.active_run().info.run_id,
+                span_context=span_context,
+                trace={},
             )
             for idx, chunk in enumerate(chunks)
         ]
 
         # Flatten the results and extract the simplification operation ID if any
-        forest = [tree for chunk in as_completed(futures) for tree in chunk.result()]
+        forest = [tree for future in as_completed(futures) for tree in future.result()[0]]
+        for future in as_completed(futures):
+            trace = future.result()[1]
         op_id = simplification_operation.get()
 
+    trace_id = mlflow.get_current_active_span()._trace_id
+    request_id = mlflow.get_current_active_span().request_id
+    trace = Trace(info=trace["info"], data=trace["data"]).from_dict(trace)
+    instance_mlflow_manager = mlflow.tracing.utils.mlflow.tracing.trace_manager.InMemoryTraceManager.get_instance()
+    span_dict = {}
+    for span in trace.data.spans:
+        span_dict[span.span_id] = LiveSpan.from_immutable_span(span, span.parent_id, request_id, trace_id)
+
+    new_trace = _Trace(info=trace.info, span_dict=span_dict)
+    instance_mlflow_manager._traces[request_id] = new_trace
```

Comment:
Here a rough idea for refactoring

```suggestion
        forest = []
        trace_manager = InMemoryTraceManager.get_instance()
        
        for future in as_completed(futures):
            trees, spans = future.result()
            forest.extend(trees)

            # Register spans from sub-process... Explain here why we need to do this
            for span in spans:
                # You may keep your convertion to LiveSpan
                trace_manager.register(span)

        op_id = simplification_operation.get()
```